src/snowflake/connector/connection.py

```diff
@@ -927,6 +931,18 @@ def __open_connection(self):
             # and what would the heartbeat frequency be
             self._add_heartbeat()
 
+        if (
+            snowflake.connector.cursor.USE_NANOARROW_CONVERTER is None
+            and snowflake.connector.cursor._SERVER_USE_NANOARROW_CONVERTER_PARAMETER
+        ) or snowflake.connector.cursor.USE_NANOARROW_CONVERTER:
```

Comment:
No strong opinion, but it feels more natural to flip the order such that we first check if `USE_NANOARROW_CONVERTER` is `True`, then check if `USE_NANOARROW_CONVERTER` is `None` and `_SERVER_USE_NANOARROW_CONVERTER_PARAMETER` is `True`?

Reply:
I feel the other way, because the default condition `USE_NANOARROW_CONVERTER` being None `_SERVER_USE_NANOARROW_CONVERTER_PARAMETER` being True, so the if condition check will pass in the first check.

anyways, this code will be removed within one month, so I'd rather keep it this way

Reply:
introduced new enum in #1758 to indicate the usage of session parameter, local settings